### ISSUE TITLE
sam build without container cannot find mvn installation

### ISSUE THREAD

**Issue by johnpauljai** [aws/aws-toolkit-jetbrains#897]
**Describe the bug**
SAM build without use-container fails in AWS Toolkit. The same command runs fine in a terminal (MacOS).

**To reproduce**
1. Configure AWS Toolkit lambda configuration to run without selecting "Build function without a container" check box.
2. Run the lambda.

**Expected behavior**
SAM build to pass and then lambda succeeds. But sam build fails with below error.

`/usr/local/bin/sam build --template /private/var/folders/70/307d2nds25b6pq7xthz1mfmw0000gp/T/template1627.yaml --build-dir /private/var/folders/70/307d2nds25b6pq7xthz1mfmw0000gp/T/lambdaBuild661`
```
2019-04-08 16:40:16 Building resource 'Function'
Traceback (most recent call last):
  File "/usr/local/bin/sam", line 11, in <module>
    load_entry_point('aws-sam-cli==0.14.2', 'console_scripts', 'sam')()
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/samcli/commands/build/command.py", line 95, in cli
    skip_pull_image, parameter_overrides)  # pragma: no cover
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/samcli/commands/build/command.py", line 133, in do_cli
    artifacts = builder.build()
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/samcli/lib/build/app_builder.py", line 99, in build
    lambda_function.runtime)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/samcli/lib/build/app_builder.py", line 190, in _build_function
    runtime)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/samcli/lib/build/app_builder.py", line 210, in _build_function_in_process
    executable_search_paths=config.executable_search_paths)
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/aws_lambda_builders/builder.py", line 110, in build
    return workflow.run()
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/aws_lambda_builders/workflow.py", line 42, in wrapper
    exec_paths = binary_path.resolver.exec_paths if not binary_path.path_provided else binary_path.binary_path
  File "/usr/local/Cellar/aws-sam-cli/0.14.2/libexec/lib/python3.7/site-packages/aws_lambda_builders/workflows/java_maven/maven_resolver.py", line 21, in exec_paths
    raise ValueError("No Maven executable found!")
ValueError: No Maven executable found!
```

**Screenshots**

**Your Environment**
 - OS: MacOS
 - JetBrains' Product: IntelliJ IDEA
 - JetBrains' Product Version: Ultimate 2019.1
 - Toolkit Version: 1.2
 - SAM CLI Version: version 0.14.2
 - JVM/Python Version: JDK 1.8

**Additional context**

**Comment by abrooksv** [comment]
It pulls it from your PATH env var, if you installed it after IDE started, you have to restart

**Comment by johnpauljai** [comment]
You were right. The path values were different in IntelliJ and Terminal. Had to follow the steps outlined here to fix it and works now. Thanks!

https://stackoverflow.com/a/26586170/1409999

**Comment by dajiangqingzhou** [comment]
I have the same error. But I install maven before installing sam cli. And I'm sure mvn command is available. 
some code in my .bashrc file:
`export M2_HOME=/home/user/apache-maven-3.6.1
export PATH=$PATH:$M2_HOME/bin:$PATH
`
@abrooksv Please Help

**Comment by abrooksv** [comment]
@dajiangqingzhou can you please open a new issue with the issue and with reproduction steps you are experiencing.

**Comment by jigar288** [comment]
I'm having the same issue. I followed this documentation (the part that says "Work with AWS Serverless Applications"). https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/key-tasks.html#key-tasks-sam

I got to the step that says "Add as Maven Project" but my IntelliJ IDEA Ultimate 2019.3 doesn't have that option. So I just tried running "sam build" and it just says "No Maven executable found!"
